src/accessors/IIterator.ts

```diff
@@ -1,6 +1,6 @@
-import { IIteratorResult } from './IIteratorResult';
-
 export interface IIterator<T> {
-    next(skip: number): IIteratorResult<T>;
+    next(packageSize?: number): Array<T>;
```

Comment:
- Great question, of which I don't have a valid answer after sleeping some nice then. I would probably rename it to `skip?: number` to allow for more functionality.
- The reason for the wrapper is due to JavaScript's Iterator Protocol, it expects the next function to return an object which has `value` and `done` properties on it. See here: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#The_iterator_protocol

> The next method always has to return an object with appropriate properties including done and value. If a non-object value gets returned (such as false or undefined), a TypeError ("iterator.next() returned a non-object value") will be thrown.